celery/result.py

```diff
@@ -969,13 +974,15 @@ def result_from_tuple(r, app=None):
     Result = app.AsyncResult
     if not isinstance(r, ResultBase):
         res, nodes = r
-        if nodes:
-            return app.GroupResult(
-                res, [result_from_tuple(child, app) for child in nodes],
-            )
-        # previously didn't include parent
         id, parent = res if isinstance(res, (list, tuple)) else (res, None)
         if parent:
             parent = result_from_tuple(parent, app)
+
+        if nodes:
+            return app.GroupResult(
```

Comment:
<img width="908" alt="screen shot 2017-08-16 at 7 31 50 pm" src="https://user-images.githubusercontent.com/2257119/29393615-a6879e90-82b9-11e7-8e6f-5afbddf3ee06.png">

first one and last four might need to be changed.